Library/Homebrew/cmd/linkapps.rb

```diff
@@ -1,22 +1,21 @@
-# Links any Applications (.app) found in installed prefixes to /Applications
 require "keg"
 require "formula"
 
 module Homebrew
   def linkapps
-    target_dir = ARGV.include?("--local") ? File.expand_path("~/Applications") : "/Applications"
+    target_dir = linkapps_target
 
-    unless File.exist? target_dir
+    unless target_dir.directory?
       opoo "#{target_dir} does not exist, stopping."
       puts "Run `mkdir #{target_dir}` first."
       exit 1
     end
 
     if ARGV.named.empty?
-      kegs = Formula.racks.map do |rack|
-        keg = rack.subdirs.map { |d| Keg.new(d) }
-        next if keg.empty?
-        keg.detect(&:linked?) || keg.max { |a, b| a.version <=> b.version }
+      kegs = Formula.installed.map do |formula|
+        formula_kegs = formula.installed_kegs
+        next if formula_kegs.empty?
+        formula_kegs.detect(&:linked?) || formula_kegs.max_by(&:version)
```

Comment:
We shouldn't use `Formula.installed` here. The thing is commands like `link/unlink/switch/uninstall/linkapp` should work fine with DIY installation, i.e. no formula required. So let's use rack directly instead of trying to load formula.